homeassistant/components/nws/__init__.py

```diff
@@ -37,6 +37,9 @@ def base_unique_id(latitude: float, longitude: float) -> str:
     return f"{latitude}_{longitude}"
 
 
+NWSConfigEntry = ConfigEntry["NWSData"]
```

Comment:
```suggestion
NWSConfigEntry = ConfigEntry[NWSData]
```

Not sure if it needs quotes here

Reply:
Yes this works, but order of execution must be swapped.  Blog entry might want to be updated to avoid future copy/paste problems.

Reply:
The quotes are needed if the type is declared later in the file.
The quotes are NOT needed if the type is declared earlier in the file, or if it is imported from another module.

HOWEVER, the recommendation is also to create type aliases after imports, but before class and methods.

So in this case, my suggestion is to keep the quotes, and move it up to line 34 (before `def base_unique_id`)

Reply:
I slightly lean towards the original suggestion to remove quotes as then the typing order follows the normal code ordering.  But, I have no stake in the ground at all.  Moved up with quotes.

Reply:
This module doesn't exist here (yet), so let's keep it simple for now and this could be considered later.

Reply:
The quotes won't be an issue once mypy adds support for PEP 695 which should be soon. It can be converted to this then:
```py
type NWSConfigEntry = ConfigEntry[NWSData]
```